Shaders/PostProcess/TemporalAntiAliasing/private/ComputeTemporalAntiAliasing.fx

```diff
@@ -12,55 +14,246 @@ cbuffer cbCameraAttribs
 {
-    return g_TextureCurrColor.Load(int3(PixelCoord, 0));
+    float3 Mean;
+    float3 Variance;
+    float3 StdDev;
+};
+
+float3 RGBToYCoCg(float3 RGB)
+{
+    float3x3 Transform = float3x3(
+         0.25, 0.50,  0.25,
+         0.50, 0.00, -0.50,
+        -0.25, 0.50, -0.25
+    );
+    return mul(Transform, RGB);
 }
 
-float4 SamplePrevColor(int2 PixelCoord)
+float3 YCoCgToRGB(float3 YCoCg)
 {
-    return g_TexturePrevColor.Load(int3(PixelCoord, 0));
+    float3x3 Transform = float3x3(
+        1.0,  1.0, -1.0,
+        1.0,  0.0,  1.0,
+        1.0, -1.0, -1.0
+    );
+    return mul(Transform, YCoCg);
+}
+
+float3 HDRToSDR(float3 Color)
+{
+    return Color * rcp(1.0 + Color);
+}
+
+float3 SDRToHDR(float3 Color)
+{
+    return Color * rcp(1.0 - Color + FLT_EPS);
+}
+
+float3 SampleCurrColor(int2 PixelCoord)
+{
+    return g_TextureCurrColor.Load(int3(PixelCoord, 0));
 }
 
-float2 SampleMotion(int2 PixelCoord)
+float3 SamplePrevColor(int2 PixelCoord)
 {
-    return g_TextureMotion.Load(int3(PixelCoord, 0)) * F3NDC_XYZ_TO_UVD_SCALE.xy;
+    return g_TexturePrevColor.Load(int3(PixelCoord, 0));
 }
 
 float SampleDepth(int2 PixelCoord)
 {
     return g_TextureDepth.Load(int3(PixelCoord, 0));
 }
 
+float2 SampleClosestMotion(int2 PixelCoord)
+{
+    float ClosestDepth = 1.0;
+    int2 ClosestOffset = int2(0, 0);
+    for (int x = -TAA_MOTION_DILATION_RADIUS; x <= TAA_MOTION_DILATION_RADIUS; x++)
+    {
+        for (int y = -TAA_MOTION_DILATION_RADIUS; y <= TAA_MOTION_DILATION_RADIUS; y++)
+        {
+            int2 Coord = int2(PixelCoord) + int2(x, y);
+            float NeighborDepth = SampleDepth(Coord);
+            if (NeighborDepth < ClosestDepth)
+            {
+                ClosestOffset = int2(x, y);
+                ClosestDepth = NeighborDepth;
+            }
+        }
+    }
+
+    return g_TextureCurrMotion.Load(int3(PixelCoord + ClosestOffset, 0)) * F3NDC_XYZ_TO_UVD_SCALE.xy;
+}
+
+float3 ClipToAABB(float3 AABBMin, float3 AABBMax, float3 P, float3 Q)
+{
+    float3 R = Q - P;
+    float3 TMin = (AABBMin - P.xyz);
+    float3 TMax = (AABBMax - P.xyz);
+
+    if (R.x > TMax.x + FLT_EPS)
+        R *= (TMax.x / R.x);
+    if (R.y > TMax.y + FLT_EPS)
+        R *= (TMax.y / R.y);
+    if (R.z > TMax.z + FLT_EPS)
+        R *= (TMax.z / R.z);
+
+    if (R.x < TMin.x - FLT_EPS)
+        R *= (TMin.x / R.x);
+    if (R.y < TMin.y - FLT_EPS)
+        R *= (TMin.y / R.y);
+    if (R.z < TMin.z - FLT_EPS)
+        R *= (TMin.z / R.z);
+
+    return P + R;
+}
+
+float ComputeDisocclusion(float2 Position)
+{
+    float2 CurrMotion = g_TextureCurrMotion.Load(int3(Position, 0)) * F3NDC_XYZ_TO_UVD_SCALE.xy;
+    float2 PrevLocation = Position.xy - CurrMotion * g_CurrCamera.f4ViewportSize.xy;
+    float2 PrevMotion = g_TexturePrevMotion.Load(int3(PrevLocation, 0)) * F3NDC_XYZ_TO_UVD_SCALE.xy;
+    float Disocclusion = length((PrevMotion - CurrMotion) * g_CurrCamera.f4ViewportSize.xy) - g_TAAAttribs.DisocclusionThreshold;
```

Comment:
Do you need to remove jitter here?

Reply:
Why do we need to remove jitter here? The velocity map should be without jiiter. We don't use a projection matrix here that could introduce jitter.